### PR TITLE
Fixes #6184 - Dynflowizes system update

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by jlsherrill** [comment]
Apologies for the delay, going to review this today.

File: app/controllers/katello/api/v2/systems_controller.rb [hunk 1]
```diff
@@ -143,8 +143,7 @@ def create
     param :host_collection_ids, Array, :desc => N_("Specify the host collections as an array")
     def update
       system_params = system_params(params)
-      @system.update_attributes!(system_params)
-      @system.refresh_subscriptions if system_params[:autoheal]
+      sync_task(::Actions::Katello::System::Update, @system, system_params)
       respond_for_update
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/systems_controller.rb]: I thought if it failed in the run phase no exception would be raised.  I'll verify this.

File: app/models/katello/glue/candlepin/consumer.rb [hunk 2]
```diff
@@ -220,8 +203,6 @@ def save_candlepin_orchestration
         case orchestration_for
         when :hypervisor
           # it's already saved = do nothing
-        when :update
-          pre_queue.create(:name => "update candlepin consumer: #{self.name}", :priority => 3, :action => [self, :update_candlepin_consumer])
         end
       end
```
Comment (jlsherrill) [inline app/models/katello/glue/candlepin/consumer.rb]: Can't we delete this whole method 'save_candlepin_orchestration' and get rid of the before_save that calls it?

File: app/models/katello/glue/pulp/consumer.rb [hunk 3]
```diff
@@ -211,10 +211,6 @@ def deactivate_pulp_node
 
       def save_pulp_orchestration
         return true if self.is_a? Hypervisor
-        case orchestration_for
-        when :update
-          pre_queue.create(:name => "update pulp consumer: #{self.name}", :priority => 3, :action => [self, :update_pulp_consumer])
-        end
       end
```
Comment (jlsherrill) [inline app/models/katello/glue/pulp/consumer.rb]: same question here.

**Comment by jlsherrill** [comment]
Seeing an issue when trying to update a system:

```
Katello::Errors::UserNotSet

unauthenticated to call a backend engine

---
- /home/jlsherri/git/katello/app/models/katello/concerns/user_extensions.rb:61:in
  `cp_oauth_header'
- /home/jlsherri/git/katello/app/lib/katello/resources/candlepin.rb:76:in `default_headers'
- /home/jlsherri/git/katello/app/lib/katello/resources/candlepin.rb:166:in `update'
- /home/jlsherri/git/katello/app/lib/actions/candlepin/consumer/update.rb:31:in `run'
- /home/jlsherri/.rvm/gems/ruby-1.9.3-p448/gems/dynflow-0.7.5/lib/dynflow/action.rb:456:in
  `block (3 levels) in execute_run'
- /home/jlsherri/.rvm/gems/ruby-1.9.3-p448/gems/dynflow-0.7.5/lib/dynflow/middleware/stack.rb:26:in
```

I think app/lib/actions/candlepin/consumer/update.rb needs to inherit from Candlepin::Abstract like some of the other candlepin actions do?

File: app/lib/actions/candlepin/consumer/refresh_subscriptions.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        input_format do
+          param :uuid, String
+        end
+
+        def plan(system, sys_params)
+          plan_self(:uuid => system.uuid) if sys_params[:autoheal]
+        end
+
+        def finalize
+          Resources::Candlepin::Consumer.refresh_entitlements(input[:uuid])
```
Comment (jlsherrill) [inline app/lib/actions/candlepin/consumer/refresh_subscriptions.rb]: I think this should be ::Katello::Resources::Candlepin::Consumer  I got a 'uninitialized constant' error.

Comment (jlsherrill) [inline app/lib/actions/candlepin/consumer/refresh_subscriptions.rb]: any reason you did this in finalize and not as part of run?

File: app/controllers/katello/api/v2/systems_controller.rb (hunk 1 above)
Comment (jlsherrill) [inline app/controllers/katello/api/v2/systems_controller.rb]: Looks like you're correct, ignore me! :+1:

File: app/lib/actions/candlepin/consumer/refresh_subscriptions.rb (hunk 4 above)
Comment (adamruzicka) [inline app/lib/actions/candlepin/consumer/refresh_subscriptions.rb]: Candlepin actions indeed need to inherit from ```Candlepin::Abstract```

File: app/lib/actions/pulp/consumer/update.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+module Actions
+  module Pulp
+    module Consumer
+      class Update < Actions::EntryAction
```
Comment (jlsherrill) [inline app/lib/actions/pulp/consumer/update.rb]: I think the same issue here, this needs to inherit from Pulp::Abstract too.  This seemed to work for me:

```
module Actions
  module Pulp
    module Consumer
      class Update < Pulp::Abstract
        input_format do
          param :uuid, String
          param :display_name, String
        end

        def plan(system)
          plan_self(:uuid => system.uuid, :display_name => system.name)
        end

        def run
          output[:response] = pulp_extensions.consumer.update(input[:uuid], :display_name => input[:name])
        end
      end
    end
  end
end
```

Comment (jlsherrill) [inline app/lib/actions/pulp/consumer/update.rb]: everything else looks good.  Feel free to squash all your commits together as you fix this and i think it'll be ready to merge.

**Comment by adamruzicka** [comment]
[test]

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
Merging, thanks @adamruzicka !  :shipit:

**Pull request merged**